src/bin/add/main.rs

```diff
@@ -141,7 +145,34 @@ fn handle_add(args: &Args) -> Result<()> {
         )?;
         update_registry_index(&url, args.quiet)?;
     }
-    let deps = &args.parse_dependencies()?;
+    let requested_features: Option<BTreeSet<&str>> = args.features.as_ref().map(|v| {
+        v.iter()
+            .map(|s| s.split(' '))
+            .flatten()
+            .filter(|s| !s.is_empty())
+            .collect()
+    });
+
+    let deps = &args.parse_dependencies(
+        requested_features
+            .as_ref()
+            .map(|s| s.iter().map(|s| s.to_string()).collect()),
+    )?;
+
+    if let Some(req_feats) = requested_features {
+        assert!(deps.len() == 1);
+        let available_features = deps[0]
+            .available_features
+            .iter()
+            .map(|s| s.trim())
+            .collect::<BTreeSet<&str>>();
+
+        let unknown_features: Vec<&&str> = req_feats.difference(&available_features).collect();
+
+        if !unknown_features.is_empty() {
+            println!("WARN: The features {:?} were requested but are not exposed by the crate. Available features are: {:?}", unknown_features, available_features);
```

Comment:
See `src/bin/upgrade/main.rs`s `deprecated_message` for a pattern to follow (including splitting it out into a function)

(sorry, I thought there were better examples in this file that would have made this clear but I was thinking of other commands)

Reply:
Also, since we'll list available features, we don't need to do it in this message.

I'd recommend keeping it direct like "Unrecognized features for {}: {:?}"

Reply:
That's true, I included it when it would error instead of warn thus the list other list wouldn't be printed, then forgot to remove it after I turned it into a warning. I'll have a look at the `deprecated_message`